File: wagerproof-ios-native/Wagerproof/Features/LearnMore/Components/LearnSlide.swift
import SwiftUI
import WagerproofDesign

/// Single carousel slide. Mirrors `wagerproof-mobile/components/learn-wagerproof/LearnSlide.tsx`.
///
/// Layout:
///   1. Glassmorphic title card (icon badge + title + description).
///   2. Custom mockup content (one of the `Slide<N>_*` views).
///   3. Optional "Why this matters" glassmorphic callout pinned beneath.
///
/// The RN version uses `BlurView` + `LinearGradient` for the glass effect. In
/// SwiftUI we lean on `.background(.ultraThinMaterial)` for the same visual,
/// with a brand-green tint gradient overlay for the value-prop card so the
/// accent reads even on light mode where `ultraThinMaterial` is near-white.
struct LearnSlide<Content: View>: View {
    let systemImage: String
    let title: String
    let description: String
    let valueProposition: String?
    @ViewBuilder let content: () -> Content

    var body: some View {
        ScrollView(showsIndicators: false) {
            VStack(spacing: Spacing.md) {
                titleCard
                content()
                if let valueProposition {
                    valueCard(valueProposition)
                }
            }
            .padding(.horizontal, Spacing.lg)
            .padding(.top, Spacing.xs)
            .padding(.bottom, Spacing.lg)
        }
    }

    // MARK: - Cards

    private var titleCard: some View {
        HStack(spacing: Spacing.md) {
            // Icon badge — gradient green pill matching RN `iconBadge`.
            ZStack {
                RoundedRectangle(cornerRadius: 10, style: .continuous)
                    .fill(
                        LinearGradient(
                            colors: [Color.appPrimary, Color.appPrimaryStrong],
                            startPoint: .topLeading,
                            endPoint: .bottomTrailing
                        )
                    )
                Image(systemName: systemImage)
                    .font(.system(size: 18, weight: .bold))
                    .foregroundStyle(Color.black)
            }
            .frame(width: 36, height: 36)

            VStack(alignment: .leading, spacing: 2) {
                Text(title)
                    .font(.system(size: 16, weight: .bold))
                    .foregroundStyle(Color.appTextPrimary)
                Text(description)
                    .font(.system(size: 12))
                    .foregroundStyle(Color.appTextSecondary)
                    .fixedSize(horizontal: false, vertical: true)
            }
            .frame(maxWidth: .infinity, alignment: .leading)
        }
        .padding(.vertical, 10)
        .padding(.horizontal, Spacing.lg)
        .background(.ultraThinMaterial)
        .clipShape(RoundedRectangle(cornerRadius: 14, style: .continuous))
        .overlay(
            RoundedRectangle(cornerRadius: 14, style: .continuous)
                .strokeBorder(Color.white.opacity(0.1), lineWidth: 1)
        )
    }

    private func valueCard(_ text: String) -> some View {
        VStack(alignment: .leading, spacing: 4) {
            HStack(spacing: 6) {
                Image(systemName: "lightbulb.fill")
                    .font(.system(size: 13, weight: .semibold))
                    .foregroundStyle(Color.appPrimary)
                Text("Why This Matters")
                    .font(.system(size: 10, weight: .bold))
                    .tracking(0.5)
                    .textCase(.uppercase)
                    .foregroundStyle(Color.appPrimary)
            }
            Text(text)
                .font(.system(size: 12))
                .lineSpacing(2)
                .foregroundStyle(Color.appTextPrimary)
                .fixedSize(horizontal: false, vertical: true)
        }
        .padding(10)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(
            LinearGradient(
                colors: [
                    Color.appPrimary.opacity(0.10),
                    Color.appPrimaryStrong.opacity(0.05),
                ],
                startPoint: .topLeading,
                endPoint: .bottomTrailing
            )
        )
        .background(.ultraThinMaterial)
        .clipShape(RoundedRectangle(cornerRadius: 12, style: .continuous))
        .overlay(
            RoundedRectangle(cornerRadius: 12, style: .continuous)
                .strokeBorder(Color.appPrimary.opacity(0.2), lineWidth: 1)
        )
    }
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/learn/LearnSlide.kt
package com.wagerproof.app.features.learn

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.wagerproof.core.design.tokens.AppColors
import com.wagerproof.core.design.tokens.Spacing

/**
 * Port of iOS `LearnSlide.swift`. Per-slide scaffold: glass title card
 * (green icon badge + title + description), custom mockup content, optional
 * "WHY THIS MATTERS" green-tinted value card.
 *
 * The iOS glass uses `.ultraThinMaterial`; Compose has no material blur, so we
 * approximate with `appSurfaceElevated` + a hairline border (established
 * translucent-surface substitution across this port).
 */
@Composable
fun LearnSlide(
    icon: ImageVector,
    title: String,
    description: String,
    valueProposition: String?,
    modifier: Modifier = Modifier,
    content: @Composable () -> Unit,
) {
    Column(
        modifier
            .verticalScroll(rememberScrollState())
            .padding(horizontal = Spacing.lg)
            .padding(top = Spacing.xs, bottom = Spacing.lg),
        verticalArrangement = Arrangement.spacedBy(Spacing.md),
    ) {
        titleCard(icon, title, description)
        content()
        valueProposition?.let { valueCard(it) }
    }
}

@Composable
private fun titleCard(icon: ImageVector, title: String, description: String) {
    Row(
        Modifier
            .clip(RoundedCornerShape(14.dp))
            .background(AppColors.appSurfaceElevated)
            .border(1.dp, Color.White.copy(alpha = 0.1f), RoundedCornerShape(14.dp))
            .padding(horizontal = Spacing.lg, vertical = 10.dp),
        horizontalArrangement = Arrangement.spacedBy(Spacing.md),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Box(
            Modifier
                .size(36.dp)
                .clip(RoundedCornerShape(10.dp))
                .background(Brush.linearGradient(listOf(AppColors.appPrimary, AppColors.appPrimaryStrong))),
            contentAlignment = Alignment.Center,
        ) {
            Icon(imageVector = icon, contentDescription = null, tint = Color.Black, modifier = Modifier.size(18.dp))
        }
        Column(verticalArrangement = Arrangement.spacedBy(2.dp)) {
            Text(title, fontSize = 16.sp, fontWeight = FontWeight.Bold, color = AppColors.appTextPrimary)
            Text(description, fontSize = 12.sp, color = AppColors.appTextSecondary)
        }
    }
}

@Composable
private fun valueCard(text: String) {
    Column(
        Modifier
            .clip(RoundedCornerShape(12.dp))
            .background(
                Brush.linearGradient(
                    listOf(AppColors.appPrimary.copy(alpha = 0.10f), AppColors.appPrimaryStrong.copy(alpha = 0.05f)),
                ),
            )
            .border(1.dp, AppColors.appPrimary.copy(alpha = 0.2f), RoundedCornerShape(12.dp))
            .padding(10.dp),
        verticalArrangement = Arrangement.spacedBy(4.dp),
    ) {
        Row(horizontalArrangement = Arrangement.spacedBy(6.dp), verticalAlignment = Alignment.CenterVertically) {
            Icon(
                imageVector = com.wagerproof.core.design.icons.AppIcon.fromSystemName("lightbulb.fill")!!.imageVector,
                contentDescription = null,
                tint = AppColors.appPrimary,
                modifier = Modifier.size(13.dp),
            )
            Text(
                "WHY THIS MATTERS",
                fontSize = 10.sp,
                fontWeight = FontWeight.Bold,
                letterSpacing = 0.5.sp,
                color = AppColors.appPrimary,
            )
        }
        Text(text, fontSize = 12.sp, color = AppColors.appTextPrimary, lineHeight = 16.sp)
    }
}
